### DISCUSSION TITLE
Introduce Tags

### DISCUSSION THREAD

**Discussion by sabatale** [filebrowser/filebrowser#989] [Ideas]
**Feature request**

**Is your feature request related to a problem? Please describe.**
Tags would make it easier to classify and search through files. At the moment the only substitute to filters we have is the file format (video, audio, etc.). Being able to tag files, such as Hip-hop, Jazz, etc. for music, would be nice.

**Describe the solution you'd like**

- New attribute in the file information named "Tags".
- These tags can be searched for (or filtered on if such a feature is added one day).
- _Bonus 1_: Values are stored and re-usable across files/folders, and ultimately can be used for reporting in some cases.
- _Bonus 2:_ Associating a color with these tags and being able to display or hide them on files in the listing.

**Describe alternatives you've considered**
Classifying types or categories through folders or, for specific formats, customize the search to add new ones (e.g. zip, etc.).

**Example (with [vue-select](https://vue-select.org/guide/values.html#tagging))**
<img src="https://user-images.githubusercontent.com/50220345/84957369-09f22280-b0c9-11ea-8594-3a6929849ef2.png"  width="343" height="200">

**Comment by hugoprudente** [comment]
That would be nice! The tags command should also be available in the command line so we could use Command Runner to auto tag images based on information that we can fetch from the command line, like part of the name is a tag, or part of the path is a tag, or a file metadata is a tag (if file metadata is not auto tagging by default)
